source/adios2/toolkit/format/bp1/BP1Writer.cpp

```diff
@@ -56,7 +56,8 @@ void BP1Writer::WriteProcessGroupIndex(
     WriteNameRecord(name, dataBuffer, dataPosition);
 
     // processID in metadata,
-    const uint32_t processID = static_cast<uint32_t>(m_BP1Aggregator.m_RankMPI);
+    const uint32_t processID =
```

Comment:
I don't recall if this caused a Windows warning, without static_cast for sure. Doesn't hurt to keep the intend of the cast. It will fail (or at least throw a warning) if uint32_t = static_cast<const uint32_t>...